forte/data/base_pack.py

```diff
@@ -481,9 +483,19 @@ def get_single(self, entry_type: Union[str, Type[EntryType]]) -> EntryType:
         Returns:
             A single data entry.
         """
-        for a in self.get(entry_type):
-            return a
-
+        idx = -1
+        for idx, a in enumerate(self.get(entry_type)):
+            if idx == payload_index:
+                return a
+        if idx < payload_index:
+            if idx == -1:
+                raise EntryNotFoundError(
+                    f"There is no {entry_type} in the provided pack."
+                )
+            raise EntryNotFoundError(
+                f"The payload index {payload_index} is larger than maximum"
+                f" {entry_type} index {idx} in the provided pack."
```

Comment:
This message is a bit hard to understand to me. Why are we comparing `payload_index` with `entry_type` index?